canister/src/heartbeat.rs

```diff
@@ -164,6 +164,17 @@ fn maybe_process_response() {
                         return;
                     }
                 }
+                for block_header_blob in response.next.iter() {
+                    let block_header = BlockHeader::consensus_decode(block_header_blob.as_slice())
+                        .expect("block header decoding must succeed");
```

Comment:
Good catch, I forgot to change this. My idea is when decoding fails, we abort adding to next_block hashes for that response, because if we continue, we may not have the previous block on which to attach.

Reply:
Done. PTAL.